pkg/kubelet/kubelet.go

```diff
@@ -2909,11 +2961,17 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod, podStatus *kubecontaine
 		}
 		// Pod allocation does not need to be updated.
 		return allocatedPod, nil
-	} else if resizable, msg := kuberuntime.IsInPlacePodVerticalScalingAllowed(pod); !resizable {
+	} else if resizable, msg := kl.containerRuntime.IsInPlacePodVerticalScalingAllowed(pod); !resizable {
 		// If there is a pending resize but the resize is not allowed, always use the allocated resources.
 		kl.recorder.Eventf(pod, v1.EventTypeWarning, events.ResizeInfeasible, msg)
 		kl.statusManager.SetPodResizeStatus(pod.UID, v1.PodResizeStatusInfeasible)
 		return allocatedPod, nil
+	} else if IsMemoryResize(pod, allocatedPod) && swapReconfigurationRequired(kl.containerRuntime, pod, allocatedPod) {
+		// If this resize involve swap set as infeasible, as IPPR with swap is not supported for beta.
+		msg = "In-place resize of pods with swap is not supported."
+		kl.recorder.Eventf(pod, v1.EventTypeWarning, events.ResizeInfeasible, msg)
+		kl.statusManager.SetPodResizeStatus(pod.UID, v1.PodResizeStatusInfeasible)
```

Comment:
Make sure to rebase when you get a chance. This will need to be updated with https://github.com/kubernetes/kubernetes/pull/130733